torch/_dynamo/variables/ctx_manager.py

```diff
@@ -189,6 +189,138 @@ def exit(self, tx, *args):
         return variables.ConstantVariable.create(None)
 
 
+class JvpIncrementNestingCtxManagerVariable(ContextWrappingVariable):
+    """represents torch.func.jvp increment/decrement nesting"""
+
+    # A guard is needed as the grad level is baked into the torch FX graph
+    # This is fine if jvp is only called from within the function
+    # being compiled. But the FX graph may be invalid in the case of a jvp
+    # call from eager that calls the compiled function, as the jvp levels
+    # may be different.
+    _guards_singleton = Guard(GlobalStateSource(), GuardBuilder.FUNCTORCH_STACK_MATCH)
+
+    @staticmethod
+    def create(tx, **kwargs):
+        var = JvpIncrementNestingCtxManagerVariable(
+            target_values=None,
+            initial_values=None,
+            **kwargs,
+        )
+        return var
+
+    def enter(self, tx):
+        install_guard(self._guards_singleton)
+        jvp_level = torch._C._functorch._jvp_increment_nesting()
+        self.set_cleanup_hook(tx, lambda: torch._C._functorch._jvp_decrement_nesting())
+        self.state.proxy = tx.output.create_node(
+            "call_function",
+            torch._C._functorch._jvp_increment_nesting,
+            (),
+            {},
+        )
+        return variables.ConstantVariable.create(jvp_level)
+
+    def exit(self, tx, *args):
+        self.state.cleanup()
+        tx.output.create_node(
+            "call_function", torch._C._functorch._jvp_decrement_nesting, (), {}
+        )
+        return variables.ConstantVariable.create(None)
+
+
+class SetFwdGradEnabledContextManager(ContextWrappingVariable):
```

Comment:
It's gotten to the point where these ctx_manager in this file are really boilerplate-y. Can we do something about it?

e.g. create a SimpleContextManager base class and then reuse it
```py
class SimpleContextManager(ContextWrappingVariable)
    def create(tx, api, target_values, **kwargs):
        return SimpleContextManager(...)
    def enter(self, tx):
        self.ctx = api()
        create_node(call_function, api, ...)
        set_exit_handler(lambda: self.ctx.__exit__())
        api.__enter__()
    def exit(self, tx):
        pass
```